src/Microsoft.AspNetCore.SignalR.Common/Internal/Formatters/TextMessageFormatter.cs

```diff
@@ -9,11 +10,18 @@ public static class TextMessageFormatter
     {
         // This record separator is supposed to be used only for JSON payloads where 0x1e character
         // will not occur (is not a valid character) and therefore it is safe to not escape it
-        internal static readonly byte RecordSeparator = 0x1e;
+        public static readonly byte RecordSeparator = 0x1e;
 
         public static void WriteRecordSeparator(Stream output)
         {
             output.WriteByte(RecordSeparator);
         }
+
+        public static void WriteRecordSeparator(IBufferWriter<byte> output)
+        {
```

Comment:
We might have to look at reducing interface calls by either making it generic or passing the pipe writer

Reply:
What we can do is call `GetMemory` once from the `IBufferWriter`, set it as a field, fill it up with bytes, and only advance when it is full or the `TextWriter` is flushed.

Reply:
If we do that for most JSON we'll call the interface twice.

Reply:
Yes, we should do that as much as we can, the code is a bit more complicated so maybe do that in a second pass after we get this in.